h2/src/main/org/h2/mvstore/tx/TransactionStore.java

```diff
@@ -730,6 +729,14 @@ void onRollback(MVMap<Object,VersionedValue> map, Object key,
                         VersionedValue existingValue, VersionedValue restoredValue);
     }
 
+    private static RollbackListener ROLLBACK_LISTENER_NONE = new RollbackListener() {
```

Comment:
`TestClearReferences` detects a problem here. May be revert this change for simplicity? Other possible tricks like `Object` field with casts or regeneration of value in a helper method are very ugly.

Reply:
making it final should fix this